aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -60,71 +175,167 @@ impl VMChangeSet {
         Ok(change_set)
     }
 
-    pub fn write_set(&self) -> &WriteSet {
-        &self.write_set
+    /// Returns a new change set, built from the storage representation. Note
+    /// that this is a rather expensive operation because it re-splits resources
+    /// from modules and is only used to support transactions with write-set
+    /// payload.
+    pub fn try_from_storage_change_set(
```

Comment:
nit:
try_ prefix and the return type seem redundant.

Reply:
Return result because there is this line:
```
checker.check_change_set(&change_set)?;
```
which checks the size of writes / bytes and errors if check does not pass.